app/script/cryptography/CryptographyRepository.coffee

```diff
@@ -393,25 +177,60 @@ class z.cryptography.CryptographyRepository
   @return [Promise] Promise that resolves with the encrypted payload
   ###
   encrypt_generic_message: (user_client_map, generic_message, payload = @_construct_payload @current_client().id) =>
-    @get_sessions user_client_map
-    .then (cryptobox_session_map) =>
-      return @_add_payload_recipients payload, generic_message, cryptobox_session_map
-
-  ###
-  Add the encrypted message for recipients to the payload message.
+    future_cipher_payloads = []
 
-  @private
-  @param payload [Object] Payload to add encrypted message for recipients to
-  @param cryptobox_sessions [Array<cryptobox.CryptoboxSession>] Sessions for all the recipients of message
-  @param generic_message [z.proto.GenericMessage] ProtoBuffer message to be send
-  @return [Object] Payload to send to backend
-  ###
-  _add_payload_recipients: (payload, generic_message, cryptobox_session_map) ->
-    for user_id, client_session_map of cryptobox_session_map
+    for user_id, client_ids of user_client_map
       payload.recipients[user_id] ?= {}
-      for client_id, cryptobox_session of client_session_map
-        payload.recipients[user_id][client_id] = @_encrypt_payload_for_session cryptobox_session, generic_message
-    return payload
+      client_ids.forEach (client_id) =>
+        session_id = @_construct_session_id user_id, client_id
+        future_cipher_payloads.push @_encrypt_payload_for_session session_id, generic_message
+
+    @logger.log "Encrypting message of type '#{generic_message.content}' for '#{Object.keys(payload.recipients).length}' users.", payload.recipients
+
+    z.util.PromiseUtil.execute_all future_cipher_payloads
+    .then (future_cipher_payloads) =>
```

Comment:
I want all error payloads together. With `Promise.all` I would only get the first error.